### PR TITLE
Adds notes about multiscale only being 2D to docs

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
Following on from request from @JulesScholler in https://github.com/napari/napari/issues/513#issuecomment-853661430 this PR adds a note about multiscale rendering only being supported in 2D right now.

Please let us know if this is more clear @JulesScholler and would have helped.

## Type of change

- [x] Documentation change

### CODE REVIEW

**Comment by sofroniewn** [comment]
We should actually also emit a warning here, will add to PR before merge!

**Comment by JulesScholler** [comment]
Thanks @sofroniewn ! Maybe you could also add that switching to 3D mode will only use the lowest resolution in the list. Which is not really pyramidal file but it is still a great feature when one wants to quickly visualize large datasets in 3D. In my case I have a lazy object for displaying 2D section but when switching to 3D it was often crashing because the dataset could be more than 1To. With this in mind I give a list of 2 lazy objects, one for the 2D section and one which will be used for 3D, avoiding the crash.

How is the resolution chosen for 2D pyramidal files?

**Comment by sofroniewn** [comment]
> Maybe you could also add that switching to 3D mode will only use the lowest resolution in the list. Which is not really pyramidal file but it is still a great feature when one wants to quickly visualize large datasets in 3D. 

I'll add something like this to the warning.

> How is the resolution chosen for 2D pyramidal files?

For 2D it is based on the zoom level the camera is at, and i believe tries to match one pixel of downsampled data to ~1 pixel on the screen (i.e. what is the resolution level that given this camera position would result in on tile pixel being on screen pixel). I'd need to do some checking to confirm if that is still the case though

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -608,7 +609,8 @@ def add_image(
             represented by a list of array like image data. If not specified by
             the user and if the data is a list of arrays that decrease in shape
             then it will be taken to be multiscale. The first image in the list
-            should be the largest.
+            should be the largest. Please note multiscale rendering is only
+            supported in 2D.
```
Comment (jni) [inline napari/components/viewer_model.py]:
```suggestion
            should be the largest. Please note multiscale rendering is only
            supported in 2D. In 3D, only the lowest resolution scale is
            displayed.
```

File: napari/components/viewer_model.py [hunk 2]
```diff
@@ -513,7 +513,8 @@ def add_image(
             Image data. Can be N >= 2 dimensional. If the last dimension has length
             3 or 4 can be interpreted as RGB or RGBA if rgb is `True`. If a
             list and arrays are decreasing in shape then the data is treated as
-            a multiscale image.
+            a multiscale image. Please note multiscale rendering is only
+            supported in 2D.
```
Comment (jni) [inline napari/components/viewer_model.py]:
```suggestion
            supported in 2D. In 3D, only the lowest resolution scale is
            displayed.
```

File: napari/layers/image/image.py [hunk 3]
```diff
@@ -39,7 +39,8 @@ class _ImageBase(IntensityVisualizationMixin, Layer):
         Image data. Can be N >= 2 dimensional. If the last dimension has length
         3 or 4 can be interpreted as RGB or RGBA if rgb is `True`. If a
         list and arrays are decreasing in shape then the data is treated as
-        a multiscale image.
+        a multiscale image. Please note multiscale rendering is only
+        supported in 2D.
```
Comment (jni) [inline napari/layers/image/image.py]:
```suggestion
        supported in 2D. In 3D, only the lowest resolution scale is
        displayed.
```

File: napari/layers/image/image.py [hunk 4]
```diff
@@ -103,15 +104,17 @@ class _ImageBase(IntensityVisualizationMixin, Layer):
         represented by a list of array like image data. If not specified by
         the user and if the data is a list of arrays that decrease in shape
         then it will be taken to be multiscale. The first image in the list
-        should be the largest.
+        should be the largest. Please note multiscale rendering is only
+        supported in 2D.
```
Comment (jni) [inline napari/layers/image/image.py]:
```suggestion
        supported in 2D. In 3D, only the lowest resolution scale is
        displayed.
```

File: napari/layers/image/image.py [hunk 5]
```diff
@@ -103,15 +104,17 @@ class _ImageBase(IntensityVisualizationMixin, Layer):
...
 
     Attributes
     ----------
     data : array or list of array
         Image data. Can be N dimensional. If the last dimension has length
         3 or 4 can be interpreted as RGB or RGBA if rgb is `True`. If a list
         and arrays are decreasing in shape then the data is treated as a
-        multiscale image.
+        multiscale image. Please note multiscale rendering is only
+        supported in 2D.
```
Comment (jni) [inline napari/layers/image/image.py]:
```suggestion
        supported in 2D. In 3D, only the lowest resolution scale is
        displayed.
```

File: napari/layers/image/image.py [hunk 6]
```diff
@@ -122,7 +125,8 @@ class _ImageBase(IntensityVisualizationMixin, Layer):
     multiscale : bool
         Whether the data is a multiscale image or not. Multiscale data is
         represented by a list of array like image data. The first image in the
-        list should be the largest.
+        list should be the largest. Please note multiscale rendering is only
+        supported in 2D.
```
Comment (jni) [inline napari/layers/image/image.py]:
```suggestion
        supported in 2D. In 3D, only the lowest resolution scale is
        displayed.
```

File: napari/layers/labels/labels.py [hunk 7]
```diff
@@ -33,7 +33,8 @@ class Labels(_ImageBase):
     Parameters
     ----------
     data : array or list of array
-        Labels data as an array or multiscale. Must be integer type or bools
+        Labels data as an array or multiscale. Must be integer type or bools.
+        Please note multiscale rendering is only supported in 2D.
```
Comment (jni) [inline napari/layers/labels/labels.py]:
```suggestion
        Please note multiscale rendering is only supported in 2D. In 3D, only
        the lowest resolution scale is displayed.
```

File: napari/layers/labels/labels.py [hunk 8]
```diff
@@ -81,18 +82,21 @@ class Labels(_ImageBase):
         represented by a list of array like image data. If not specified by
         the user and if the data is a list of arrays that decrease in shape
         then it will be taken to be multiscale. The first image in the list
-        should be the largest.
+        should be the largest. Please note multiscale rendering is only
+        supported in 2D.
```
Comment (jni) [inline napari/layers/labels/labels.py]:
```suggestion
        supported in 2D. In 3D, only the lowest resolution scale is
        displayed.
```

File: napari/layers/labels/labels.py [hunk 9]
```diff
@@ -81,18 +82,21 @@ class Labels(_ImageBase):
...
     ----------
-    data : array
-        Integer label data. Can be N dimensional. Every pixel contains
-        an integer ID corresponding to the region it belongs to. The label 0 is
-        rendered as transparent.
+    data : array or list of array
+        Integer label data as an array or multiscale. Can be N dimensional.
+        Every pixel contains an integer ID corresponding to the region it
+        belongs to. The label 0 is rendered as transparent. Please note
+        multiscale rendering is only supported in 2D.
```
Comment (jni) [inline napari/layers/labels/labels.py]:
```suggestion
        multiscale rendering is only supported in 2D. In 3D, only
        the lowest resolution scale is displayed.
```

File: napari/layers/labels/labels.py [hunk 10]
```diff
@@ -81,18 +82,21 @@ class Labels(_ImageBase):
...
     multiscale : bool
         Whether the data is a multiscale image or not. Multiscale data is
         represented by a list of array like image data. The first image in the
-        list should be the largest.
+        list should be the largest. Please note multiscale rendering is only
+        supported in 2D.
```
Comment (jni) [inline napari/layers/labels/labels.py]:
```suggestion
        supported in 2D. In 3D, only the lowest resolution scale is
        displayed.
```

**Comment by sofroniewn** [comment]
I have now added a warning message so the user will see this too. This is ready for review @jni

**Review by tlambert03 — APPROVED** [review]

**Pull request merged**